File: src/db/exportDB.ts
import * as FileSystem from "expo-file-system";
import * as Sharing from "expo-sharing";
import { DB_NAME } from "./database";

export async function shareDatabase() {
    try {
        const dbUri =
            (FileSystem as any).documentDirectory + "SQLite/" + DB_NAME;
        const info = await FileSystem.getInfoAsync(dbUri);
        if (!info.exists) {
            alert("❌ ملف قاعدة البيانات غير موجود!");
            return false;
        }

        await Sharing.shareAsync(dbUri);
        return true;
    } catch (e) {
        console.log("Share Error:", e);
        alert("حدث خطأ أثناء مشاركة قاعدة البيانات.");
        return false;
    }
}

File: lib/db/export_db.dart
import 'dart:io';

import 'package:path/path.dart';
import 'package:path_provider/path_provider.dart';
import 'package:share_plus/share_plus.dart';
import 'database.dart';

Future<bool> shareDatabase() async {
  try {
    final documentsDirectory = await getApplicationDocumentsDirectory();
    final dbPath = join(documentsDirectory.path, DB_NAME);

    final file = File(dbPath);
    if (!await file.exists()) {
      return false;
    }

    await Share.shareXFiles([XFile(dbPath)], text: 'Database export');
    return true;
  } catch (e) {
    print('Share Error: $e');
    return false;
  }
}
